velox/exec/Window.cpp

```diff
@@ -41,6 +41,35 @@ void initKeyInfo(
   }
 }
 
+std::optional<column_index_t> fieldArgToChannel(
+    const core::TypedExprPtr arg,
+    const RowTypePtr& inputType) {
+  if (arg) {
+    std::optional<column_index_t> argChannel =
+        exprToChannel(arg.get(), inputType);
+    VELOX_CHECK(
+        argChannel.value() != kConstantChannel,
+        "Window doesn't allow constant arguments or frame end-points");
+    return argChannel;
+  }
+  return std::nullopt;
+}
+
+void fixOutOfBoundFrames(
```

Comment:
You can move this code in an anonymous namespace closer to where it is used for better readability.

Reply:
Sure, thanks for the suggestion!